torch/backends/cuda/__init__.py

```diff
@@ -304,12 +304,30 @@ def can_use_efficient_attention(params: SDPAParams, debug: bool = False) -> bool
     """
     return torch._C._can_use_mem_efficient_attention(params, debug)
 
+def cudnn_sdp_enabled():
+    r"""
+    .. warning:: This flag is beta and subject to change.
+
+    Returns whether cuDNN scaled dot product attention is enabled or not.
+    """
+    return torch._C._get_cudnn_sdp_enabled()
+
+
+def enable_cudnn_sdp(enabled: bool):
+    r"""
+    .. warning:: This flag is beta and subject to change.
+
+    Enables or disables math scaled dot product attention.
+    """
+    torch._C._set_sdp_use_cudnn(enabled)
+
 
 @contextlib.contextmanager
 def sdp_kernel(
     enable_flash: bool = True,
     enable_math: bool = True,
     enable_mem_efficient: bool = True,
+    enable_cudnn: bool = True,
```

Comment:
I kinda want to deprecate this function and change it to have a different api: https://github.com/pytorch/pytorch/pull/114689

But need to figure out how best to do this. 

All that being said this is the right way to do it for now